Gems/AtomLyIntegration/CommonFeatures/Code/Source/PostProcess/ColorGrading/HDRColorGradingComponentController.cpp

```diff
@@ -0,0 +1,160 @@
+            return m_configuration.MemberName;                                                                                                 \
+        }                                                                                                                                      \
+        void HDRColorGradingComponentController::Set##Name(ValueType val)                                                                     \
+        {                                                                                                                                      \
+            if (m_settingsInterface)                                                                                                           \
+            {                                                                                                                                  \
+                m_settingsInterface->Set##Name(val);                                                                                           \
+                m_settingsInterface->OnConfigChanged();                                                                                        \
+                m_configuration.MemberName = m_settingsInterface->Get##Name();                                                                 \
+            }                                                                                                                                  \
+            else                                                                                                                               \
+            {                                                                                                                                  \
+                m_configuration.MemberName = val;                                                                                              \
+            }                                                                                                                                  \
+        }
+
+#define AZ_GFX_COMMON_OVERRIDE(ValueType, Name, MemberName, OverrideValueType)                                                             \
+        OverrideValueType HDRColorGradingComponentController::Get##Name##Override() const                                                     \
+        {                                                                                                                                      \
+            return m_configuration.MemberName##Override;                                                                                       \
+        }                                                                                                                                      \
+        void HDRColorGradingComponentController::Set##Name##Override(OverrideValueType val)                                                   \
```

Comment:
Should value type be const& in all of the function parameters and returns?

Reply:
Good call! Let me remove the override macro here as I do not use it.

Reply:
The inherited interface is using a macro definition with non-const& virtual functions. I'd like to keep this for now.